### PR TITLE
Device Update for IoT Hub GA release updates

### PR DESCRIPTION
**Pull request by dpokluda**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/deviceupdate/azure-iot-deviceupdate/swagger/README.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ autorest --java --use=C:/work/autorest.java
 
 ## Generate autorest code
 ```yaml
-input-file: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/b9b91929c304f8fb44002267b6c98d9fb9dde014/specification/deviceupdate/data-plane/Microsoft.DeviceUpdate/preview/2022-07-01-preview/deviceupdate.json
+input-file: https://raw.githubusercontent.com/dpokluda/azure-rest-api-specs/bbf13b0366e5863ce65f84d588ed5681326af0cb/specification/deviceupdate/data-plane/Microsoft.DeviceUpdate/stable/2022-10-01/deviceupdate.json
```
Comment (weidongxu-microsoft) [inline sdk/deviceupdate/azure-iot-deviceupdate/swagger/README.md:23]: I think you need to switch to Azure main branch (maybe another regen), before PR can be merged.

Comment (weidongxu-microsoft) [reply sdk/deviceupdate/azure-iot-deviceupdate/swagger/README.md:23]: If you have swagger PR pending, let us know the link and state of it (whether it is approved, or there is change request on it that likely would result in SDK change).

Comment (dpokluda) [reply sdk/deviceupdate/azure-iot-deviceupdate/swagger/README.md]: Updated.

**Comment by weidongxu-microsoft** [comment]
@haolingdong-msft Could you go one round of review, then add Srikanta if it looks good to you?

Since there is no change in ApiView, I think it should be quick for Srikanta.

**Comment by weidongxu-microsoft** [comment]
I've merged the CI change https://github.com/Azure/azure-sdk-for-java/pull/30236

Next time you release azure-iot-deviceupdate, you will see it automatically checked in right panel (and azure-resourcemanager-deviceupdate by default unchecked). This should by default be the correct configure for releasing data-plane SDK.

File: sdk/deviceupdate/azure-iot-deviceupdate/CHANGELOG.md [hunk 2]
```diff
@@ -3,12 +3,11 @@
...
 ### Features Added
-
-### Breaking Changes
-
-### Bugs Fixed
+- Added filter to DeviceManagementOperations.listDeviceClasses method
+- Updated description for some methods to be more descriptive and less ambiguous
 
 ### Other Changes
+- Removed filter from DeviceManagementOperations.listBestUpdatesForGroup method
```
Comment (haolingdong-msft) [inline sdk/deviceupdate/azure-iot-deviceupdate/CHANGELOG.md:10]: Add dependency updates similar to previous releases:

#### Dependency Updates
- Upgraded `azure-core` from `1.30.0` to version `1.31.0`.
- Upgraded `azure-core-http-netty` from `1.12.3` to version `1.12.4`.

[the remark this answers is not in the document]
Comment (srnagar) [reply sdk/deviceupdate/azure-iot-deviceupdate/CHANGELOG.md:10]: This is a breaking change. So, we should add this to the "Breaking Changes" section.

[the remark this answers is not in the document]
Comment (haolingdong-msft) [reply sdk/deviceupdate/azure-iot-deviceupdate/CHANGELOG.md:10]: Also it is suggested to change the content to `Removed optional "filter" query parameter from listBestUpdatesForGroup method in DeviceManagementClient and DeviceManagementAsyncClient`.

[the remark this answers is not in the document]
Comment (dpokluda) [reply sdk/deviceupdate/azure-iot-deviceupdate/CHANGELOG.md]: Fixed

**Review by weidongxu-microsoft — APPROVED** [review]
LGTM

Let us know your release plan, as Sept. monthly release is starting.

**Review by haolingdong-msft — APPROVED** [review]

File: eng/versioning/version_client.txt [hunk 3]
```diff
@@ -101,7 +101,7 @@ com.azure:azure-digitaltwins-core;1.3.0;1.4.0-beta.1
 com.azure:azure-e2e;1.0.0-beta.1;1.0.0-beta.1
 com.azure:azure-identity;1.5.4;1.6.0-beta.1
 com.azure:azure-identity-perf;1.0.0-beta.1;1.0.0-beta.1
-com.azure:azure-iot-deviceupdate;1.0.0-beta.3;1.0.0-beta.4
+com.azure:azure-iot-deviceupdate;1.0.0-beta.4;1.0.0
```
Comment (weidongxu-microsoft) [inline eng/versioning/version_client.txt:104]: It should be `com.azure:azure-iot-deviceupdate;1.0.0-beta.3;1.0.0`

First version is one of the last released, second version is the one that to be released next.

Comment (dpokluda) [reply eng/versioning/version_client.txt]: Thank you.

**Review by srnagar — APPROVED** [review]

File: sdk/deviceupdate/azure-iot-deviceupdate/CHANGELOG.md [hunk 4]
```diff
@@ -1,15 +1,21 @@
 # Release History
 
-## 1.0.0-beta.4 (Unreleased)
+## 1.0.0 (Unreleased)
```
Comment (srnagar) [inline sdk/deviceupdate/azure-iot-deviceupdate/CHANGELOG.md:3]: nit: update the release date. The release pipeline will block the release if the date is invalid.
```suggestion
## 1.0.0 (2022-09-02)
```

**Review by srnagar — APPROVED** [review]

**Pull request merged**